tests/warn/redundant-null.scala

```diff
@@ -0,0 +1,41 @@
+class A
+class B
+class C
+
+val n = null
+
+def f(s: A) = s match
+  case _: n.type =>
+  case _: A =>
+  case _: n.type => // warn
```

Comment:
Can you also add the warning name in the test code? like `// warn: unreachable` and `// warn: null only`, etc.

Reply:
Done